Tangem/Modules/ExpressTokensList/ExpressTokensListViewModel.swift

```diff
@@ -8,140 +8,190 @@
     @Published var availableTokens: [ExpressTokenItemViewModel] = []
     @Published var unavailableTokens: [ExpressTokenItemViewModel] = []
 
     var unavailableSectionHeader: String {
-        Localization.exchangeTokensUnavailableTokensHeader("Bitcoin")
+        Localization.exchangeTokensUnavailableTokensHeader(initialWalletType.name)
+    }
+
+    var isEmptyView: Bool {
+        searchText.isEmpty && availableTokens.isEmpty && unavailableTokens.isEmpty
     }
 
     // MARK: - Dependencies
 
+    private let initialWalletType: InitialWalletType
+    private let walletModels: [WalletModel]
+    private let expressAPIProvider: ExpressAPIProvider
+    private unowned let expressInteractor: ExpressInteractor
     private unowned let coordinator: ExpressTokensListRoutable
+
+    // MARK: - Internal
+
+    private var availableWalletModels: [WalletModel] = []
+    private var unavailableWalletModels: [WalletModel] = []
     private var bag: Set<AnyCancellable> = []
 
-    init(coordinator: ExpressTokensListRoutable) {
+    init(
+        initialWalletType: InitialWalletType,
+        walletModels: [WalletModel],
+        expressAPIProvider: ExpressAPIProvider,
+        expressInteractor: ExpressInteractor,
+        coordinator: ExpressTokensListRoutable
+    ) {
+        self.initialWalletType = initialWalletType
+        self.walletModels = walletModels.filter { $0 != initialWalletType.wallet }
+        self.expressAPIProvider = expressAPIProvider
+        self.expressInteractor = expressInteractor
         self.coordinator = coordinator
-        setupView()
+
         bind()
+        loadView()
     }
 }
 
 // MARK: - Private
 
 private extension ExpressTokensListViewModel {
+    func loadView() {
+        runTask(in: self) { viewModel in
+            let availablePairs = try await viewModel.loadAvailablePairs()
+            viewModel.updateWalletModels(availableCurrencies: availablePairs)
+        }
+    }
+
+    func loadAvailablePairs() async throws -> [ExpressCurrency] {
+        var currencies = walletModels.map { $0.currency }
+
+        switch initialWalletType {
+        case .source(let wallet):
+            let pairs = try await expressAPIProvider.pairs(from: [wallet.currency], to: currencies)
+            return pairs.map { $0.destination }
+        case .destination(let wallet):
+            let pairs = try await expressAPIProvider.pairs(from: currencies, to: [wallet.currency])
+            return pairs.map { $0.source }
+        }
+    }
+
+    func updateWalletModels(availableCurrencies: [ExpressCurrency]) {
+        availableWalletModels.removeAll()
+        unavailableWalletModels.removeAll()
+
+        walletModels.forEach { walletModel in
+            let isAvailable = availableCurrencies.contains { walletModel.currency == $0 }
+            if isAvailable {
+                availableWalletModels.append(walletModel)
+            } else {
+                unavailableWalletModels.append(walletModel)
+            }
+        }
```

Comment:
будет быстрее так проверять
```suggestion
        let currenciesSet = availableCurrencies.toSet()
        walletModels.forEach { walletModel in
            if currenciesSet.contains(walletModel.currency) {
                availableWalletModels.append(walletModel)
            } else {
                unavailableWalletModels.append(walletModel)
            }
        }
```

Reply:
Я вообще хотел в словарь конечно складывать, но там сортировка теряется, нам еще обсудить надо)

Reply:
✅